src/jdk.jpackage/unix/classes/jdk/jpackage/internal/PackageScripts.java

```diff
@@ -78,14 +78,20 @@ void saveInFolder(Path folder) throws IOException {
 
     static class ResourceConfig {
 
-        ResourceConfig(String defaultName, String categoryId) {
+        ResourceConfig(String defaultName, String categoryId, boolean noDefault) {
```

Comment:
Don't you think `String defaultName, String categoryId, boolean noDefault` signature looks odd? You have a parameter specifying the default name and another parameter that specifies if the default name is set or not.

Wouldn't:
```
ResourceConfig(Optional<String> defaultName, String categoryId)
```
be less confusing?

For backward compatibility keep the old ctor, but redefine it:
```
ResourceConfig(String defaultName, String categoryId) {
    this(Optional.of(defaultName), categoryId);
}
```

Reply:
In your suggested case how `ResourceConfig.getDefaultPublicName()` will figure out public name for resource if `defaultName` is not provided? Maybe `String resourceName, String categoryId, boolean noDefault` will be less confusing?

Reply:
I will add `ResourceConfig(Optional<String> defaultName, Optional<String> publicName, String categoryId)` instead of suggested.